Add a Makefile to make pylinting easier.

fxdgear opened the pull request:
Also, Fix a couple pylint errors.

1. ignore pointless statement, it's not pointless cause these lines populate a memoprop.
2. Move None to the right spot

closes #21

fxdgear commented on Makefile:
```diff
@@ -0,0 +1,27 @@
+run-pytest: build
+	podman exec -ti -w/carthage_aws $(CONTAINER_NAME) && \
+		pytest-3 -v && \
+		--carthage-config=.github/${USER}_test_config.yml
```
I do wonder a little about this. I think having a `nick_test_config.yml` or who ever is running things to be able to run the tests locally is a good thing. But I don't know if you have a prefered way or location or naming convention for the local test config.

hartmans reviewed (changes requested):
I don't think we want developers to need a container to use this.
I think it is reasonable for a developer's environment to satisfy pylint as a requirement, and if they want a container they can build that.
I'd expect a README that includes a example pip command to install the local project's development requirements.
I wouldn't actually have a makefile target for this, because the developer may need to change the command depending on whether they are in a venv, using a container, or needing to add a flag like --break-system-packages.
(or are mostly going to satisfy from OS packages).

I would expect two makefile targets as part of the makefile interface:
* a target to install the git hooks
* A target to a makefile target to run the pre-commit checks, that can be run by the hooks or manually.

I don't know how to handle carthage_base.
I think it is reasonable for carthage to be a development requirement of carthage_aws.

I do think it is reasonable for the github actions to use a container, and that will do a lot to isolate us from the runner's old and static python.

fxdgear commented:
Ok I've added docs to the readme to address a few ways to install dependencies. 

1. can install things using poetry and it's virtualenv. 
2. can install things using pip with --break-system-packages
3. can install OS managed packages 

Addressed the way pylint is run in the pre-commit hook. It'll first check to see if `pytlint` is installed on the path. If you're in a venv it'll get installed with the dev requirements. If you're installing system level packages it was addressed to install pylint there. If pylint isn't found then it'll default to using the container. 

I added docs on how to install the githooks. 

I also added docs on how to run the tests either locally or in a container.
